src/main/java/Duke.java

```diff
@@ -1,10 +1,313 @@
+            Task task = tasks[i];
+            printStatement(String.format("%d.%s", i + 1, task));
+        }
+    }
+
+    /**
+     * Lists all tasks.
+     */
+    private static void listTasks() {
+        if (numberOfTasks == 0) {
+            printStatements(LIST_NO_TASK_MESSAGE);
+        } else {
+            printHorizontalLine();
+            printAllTasks();
+            printHorizontalLine();
+        }
+    }
+
+    /**
+     * Marks a task as done based on the order of the list.
+     *
+     * @param commandArgs this should contain the task number to mark as done
+     */
+    private static void markTaskDone(String commandArgs) {
+        try {
+            String argValue = parseArgument(commandArgs, null);
+            if (argValue == null) {
+                throw new Exception(ERROR_INVALID_TASK_NUMBER_MESSAGE);
+            }
+            int taskNumber = Integer.parseInt(argValue);
+            if (taskNumber > 0 && taskNumber <= numberOfTasks) {
+                int taskIndex = taskNumber - 1;
+                tasks[taskIndex].isDone = true;
+                Task task = tasks[taskIndex];
+                printStatements(TASK_MARK_AS_DONE_FORMAT,
+                        String.format(DOUBLE_SPACE_PREFIX_STRING_FORMAT, task));
+            } else {
+                throw new Exception(ERROR_INVALID_TASK_NUMBER_MESSAGE);
+            }
+        } catch (Exception e) {
+            printError(ERROR_INVALID_TASK_NUMBER_MESSAGE);
+        }
+    }
+
+    private static void printHorizontalLine() {
+        System.out.println("\t" + HORIZONTAL_LINE);
+    }
+
+    /**
+     * Prints a statement with a tab and spacing character.
+     * Can print multiple lines if the statement string has line
+     * breaks in it.
+     */
+    private static void printStatement(String statement) {
+        String[] fragments = statement.split("\\R");
+        for (String fragment : fragments) {
+            System.out.println("\t " + fragment);
+        }
+    }
+
+    /**
+     * Prints statements within horizontal line borders.
+     */
+    private static void printStatements(String... statement) {
+        printHorizontalLine();
+        for (String s : statement) {
+            printStatement(s);
+        }
+        printHorizontalLine();
+    }
+
+    private static void printError(String errorDescription) {
+        printHorizontalLine();
+        printStatement(String.format("%s %s", ERROR_PREFIX_MESSAGE, errorDescription));
+        printHorizontalLine();
+    }
+
+    private static void printWelcomeMessage() {
+        printStatements(LOGO, WELCOME_MESSAGE);
+    }
+
+    private static void exitProgram() {
+        printStatements(EXIT_MESSAGE);
+        System.exit(0);
+    }
+
+    /**
+     * Parses the user input and attempts to execute the command
+     * with the arguments.
+     */
+    private static void executeCommand(String userInput) {
+        String[] parsedCommand = parseCommand(userInput);
```

Comment:
Plural form should be used on names representing a collection of objects.

Reply:
I will modify the variable name, thanks!